### ISSUE TITLE
Callback not getting called from ClientNetworkMonitor cellularDidChange method

### ISSUE THREAD

**Issue by tikekar** [grpc/grpc-swift#530]
### New Issue Checklist

- [ ] Updated grpc-swift to the latest version
- [ X] I read the [Contribution Guidelines](https://github.com/grpc/grpc-swift/blob/master/CONTRIBUTING.md)
- [X ] I searched for [existing GitHub issues](https://github.com/grpc/grpc-swift/issues)(especially, there are many [resolved issues related to SSL](https://github.com/grpc/grpc-swift/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed+ssl))

### Issue Description
When I turn the airplane mode on and off from wifi, the cellularDidChange method of ClientNetworkMonitor class gets called. But the callback "self.callback(State..." doesn't get called. If I remove the if() statement surrounding it, it gets called.
```
// commenting the if statement works.
// if let newCellularName = newCellularName, self.cellularName != newCellularName {
        self.cellularName = newCellularName
        self.callback(State(lastChange: .cellularTechnology(technology: newCellularName ?? ""),
                            isReachable: self.isReachable ?? false))
      // }
```

##### Complete output when running grpc-swift, including the stack trace and command used

<details>
  <pre>[INSERT OUTPUT HERE]</pre>
</details>

### Environment

<details>
  <pre>
  
| Key                           | Value                                                         |
| ----------------------------- | --------------------------------------------------------------|
| OS Version                    |   x.x.x                                                       |
| Swift Version                 |   x.x.x                                                       |
| Xcode Version                 |   x.x.x                                                       |
| gRPC-Swift Version            |   x.x.x                                                       |
| gRPC-Swift Version            |   x.x.x                                                       |
| protoc Version                |   x.x.x                                                       |
| protoc-gen-swift Version      |   x.x.x                                                       |
| protoc-gen-swiftgrpc Version  |   x.x.x                                                       |
  </pre>
</details>

**Comment by rebello95** [comment]
Thanks for filing, @tikekar. What problems is this causing for your application? This API is supposed to be used for restarting a channel when we know it needs to be switched to another reachable connection. In this case, there are no reachable connections.

From what I've seen in past experimentation, disabling airplane mode will:
- allow the connection to automatically resume if it hasn't timed out
- receive data that was missed while airplane mode was enabled (if the connection hasn't timed out)
- send a notification through this interface notifying you of a new connection

Are you not seeing this behavior?

**Comment by tikekar** [comment]
My use case with ClientNetworkMonitor so far is -
* Whenever reachability changes, I want to call the callback and notify user whether internet is available or not.

Not sure if I am missing here something but can you help answer following questions?
Questions -
* If there are no reachable connections, what happens to the channel then? when I put my phone on airplane mode, I was assuming that I should be able to call the self.callback. 
* I was little confused regarding when should the reachabilityDidChange get called and when should cellularDidChange get called? First I thought that reachabilityDidChange get triggered when I set the airplane mode. But instead it calls cellularDidChange for me.

Following is my code for ClientNetworkMonitor initialization:
```
self.clientNetworkMonitor = ClientNetworkMonitor(callback: { (state) in
      
            if state.isReachable {
                 NotificationCenter.default.post(name: NSNotification.Name(rawValue: kNetworkStateChanged), object: NetworkState.connected)
            }
            else {
                NotificationCenter.default.post(name: NSNotification.Name(rawValue: kNetworkStateChanged), object: NetworkState.noInternet)
            }
        })
```

**Comment by rebello95** [comment]
> If there are no reachable connections, what happens to the channel then? when I put my phone on airplane mode, I was assuming that I should be able to call the self.callback.

I just ran a test locally, and on my device I'm seeing the following output when starting the app and toggling airplane mode on/off:

```
// Launch with cellular + wifi enabled
Changed: cellularTechnology(technology: "CTRadioAccessTechnologyLTE"), isReachable: true
// Turn on airplane mode
Changed: reachability(isReachable: false), isReachable: false
Changed: reachability(isReachable: false), isReachable: false
Changed: cellularTechnology(technology: "CTRadioAccessTechnologyLTE"), isReachable: false
// Turn off airplane mode
Changed: cellularTechnology(technology: "CTRadioAccessTechnologyLTE"), isReachable: false
Changed: reachability(isReachable: true), isReachable: true
Changed: reachability(isReachable: true), isReachable: true
Changed: reachability(isReachable: true), isReachable: true
```

Based on this, it seems like you should be getting notified when airplane mode turns on and off - is that not the case?

**Comment by tikekar** [comment]
Yes for me somehow when I change the airplane mode ON, it calls cellularDidChange only. Anyway currently our grpc connections work only on our company's wifi as we are still working on it and not released it yet. Thus I am still working on its full testing. But one question - 
Does the isReachable flag only tied to internet connection (network) availability? Or the isReachable can be false when the connect stream drops or channel drops for any other reason?

**Comment by rebello95** [comment]
> Does the isReachable flag only tied to internet connection (network) availability? Or the isReachable can be false when the connect stream drops or channel drops for any other reason?

This only applies to network reachability. If the channel drops for another reason, the channel should shut down / time out.
